src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStep.java

```diff
@@ -171,11 +181,11 @@ protected Void run () throws Exception {
             FilePath ws = getContext().get(FilePath.class);
             assert ws != null;
             FilePath path = ws.child(step.getFile());
-            if (path.exists()) {
-                throw new FileAlreadyExistsException(path.getRemote() + " already exist.");
-            }
             if (path.isDirectory()) {
-                throw new FileNotFoundException(path.getRemote() + " is a directory.");
+                throw new FileAlreadyExistsException(path.getRemote(), null, "is a directory");
```

Comment:
Change of behavior can cause people's pipelines to stop working.
So it at least needs to be documented or changed back to a `FileNotFoundException`.

Reply:
Yep, couldn't resist, as it made no sense to me to mix oio and nio exceptions. But will revert it to not introduce breaking changes.